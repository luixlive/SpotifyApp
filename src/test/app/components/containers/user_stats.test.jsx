import configureStore from 'redux-mock-store';
import { MemoryRouter, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

import {
  ConnectedUserStats,
  UserStats,
} from './../../../../app/components/containers/user_stats';
import emptyFunction from './../../../test_utils/empty_function';
import initialState from './../../../test_utils/initial_state';

describe('App Components - UserStats', () => {
  describe('Snapshots', () => {
    let props;
    beforeAll(() => {
      props = {
        isUserAuthenticated: false,
        loadUserFinished: false,
        loadUserStats: emptyFunction,
      };
    });

    it('renders user not authenticated and load not finished', () => {
      const rendered = renderer.create(<UserStats {...props} />).toJSON();
      expect(rendered).toMatchSnapshot();
    });

    it('renders user authenticated and load not finished', () => {
      props.isUserAuthenticated = true;
      const rendered = renderer.create(<UserStats {...props} />).toJSON();
      expect(rendered).toMatchSnapshot();
    });

    it('renders user authenticated and load finished', () => {
      props.loadUserFinished = true;
      const rendered = renderer
        .create(<UserStats {...props} />)
        .toJSON();
      expect(rendered).toMatchSnapshot();
    });

    it('renders user not authenticated and load finished', () => {
      props.isUserAuthenticated = false;
      // With Route we mock the current location of the router
      const rendered = renderer
        .create((
          <MemoryRouter>
            <Route component={() => <UserStats {...props} />} path="/stats" />
          </MemoryRouter>))
        .toJSON();
      expect(rendered).toMatchSnapshot();
    });
  });

  describe('Behavior', () => {
    describe('General', () => {
      let props;
      let wrapper;
      beforeAll(() => {
        props = {
          isUserAuthenticated: false,
          loadUserFinished: false,
          loadUserStats: emptyFunction,
        };
        wrapper = shallow(<UserStats {...props} />);
      });

      it('renders', () => {
        expect(wrapper.length).toEqual(1);
      });

      it('renders big loader', () => {
        expect(wrapper.find('Loader').length).toEqual(1);
        expect(wrapper.find('Loader').prop('size')).toEqual('big');
      });

      it('calls loadUserStats when componentDidMount', () => {
        const spy = jest.fn();
        mount(<UserStats {...props} loadUserStats={spy} />);
        expect(spy).toHaveBeenCalledTimes(1);
      });
    });
  });

  describe('Provider', () => {
    const mockStore = configureStore();
    let store;
    let wrapper;
    beforeEach(() => {
      store = mockStore(initialState);
      wrapper = mount((
        <Provider store={store}>
          <ConnectedUserStats />
        </Provider>
      ));
    });

    it('renders', () => {
      expect(wrapper.find(ConnectedUserStats).length).toEqual(1);
    });

    it('matches initial state', () => {
      expect(wrapper.find(UserStats).prop('loadUserStats'))
        .toBeInstanceOf(Function);
    });
  });
});
