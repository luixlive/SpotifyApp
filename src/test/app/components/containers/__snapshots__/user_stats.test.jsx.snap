// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Components - UserStats Snapshots renders user authenticated and load finished 1`] = `
<div
  className="ui basic segment body"
>
  <div
    className="ui active transition visible inverted dimmer"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="center"
      >
        <div
          className="ui big inverted loader"
        />
      </div>
    </div>
  </div>
  User stats
</div>
`;

exports[`App Components - UserStats Snapshots renders user authenticated and load not finished 1`] = `
<div
  className="ui basic segment body"
>
  <div
    className="ui active transition visible inverted dimmer"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="center"
      >
        <div
          className="ui big inverted loader"
        />
      </div>
    </div>
  </div>
  User stats
</div>
`;

exports[`App Components - UserStats Snapshots renders user not authenticated and load finished 1`] = `null`;

exports[`App Components - UserStats Snapshots renders user not authenticated and load not finished 1`] = `
<div
  className="ui basic segment body"
>
  <div
    className="ui active transition visible inverted dimmer"
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="center"
      >
        <div
          className="ui big inverted loader"
        />
      </div>
    </div>
  </div>
  User stats
</div>
`;
